stacker/providers/aws/default.py

```diff
@@ -106,18 +111,43 @@ def cloudformation(self):
 
         return self._cloudformation
 
+    @property
+    def listener(self):
+        # deals w/ multiprocessing issues w/ sharing ssl conns
+        # see https://github.com/remind101/stacker/issues/196
+        pid = os.getpid()
+        if pid != self._pid or not self._listener:
+            session = get_session(self.region)
+            self._listener = CloudListener(
+                LISTENER_NAME, session=session)
+            self._listener.start()
+
+        return self._listener
+
+    def pre_run(self):
+        logger.debug("Clearing old sqs queue")
+        messages = self.listener.get_messages()
+        for message in messages:
+            message.delete()
+        logger.info("Done clearing queue")
+
+    def parse_stack_events(self):
+        messages = self.listener.get_messages()
+        for message in messages:
+            message_dict = message.__dict__
+            message_dict['fqn'] = message_dict['StackName']
+            self._stacks[message.StackName] = message_dict
+            message.delete()
+
     def get_stack(self, stack_name, **kwargs):
-        try:
-            return retry_on_throttling(
-                self.cloudformation.describe_stacks,
-                kwargs=dict(StackName=stack_name))['Stacks'][0]
-        except botocore.exceptions.ClientError as e:
-            if "does not exist" not in e.message:
-                raise
+        self.parse_stack_events()
+        if stack_name in self._stacks:
```

Comment:
Rather than testing if the name is in the stack, just try to get it - and deal with the exception as it raises:

```
try:
    return self._stacks[stack_name]
except KeyError:
    raise exceptions.StackDoesNotExist(stack_name)
```